packages/rpc-methods/src/restricted/getBip32Entropy.ts

```diff
@@ -0,0 +1,260 @@
+  GetBip32EntropySpecification
+> = ({
+  allowedCaveats = null,
+  methodHooks,
+}: GetBip32EntropySpecificationBuilderOptions) => {
+  return {
+    permissionType: PermissionType.RestrictedMethod,
+    targetKey,
+    allowedCaveats: [
+      SnapCaveatType.PermittedDerivationPaths,
+      ...(allowedCaveats ?? []),
+    ],
+    methodImplementation: getBip32EntropyImplementation(methodHooks),
+    validator: ({ caveats }) => {
+      if (
+        caveats?.length !== 1 ||
+        caveats[0].type !== SnapCaveatType.PermittedDerivationPaths
+      ) {
+        throw ethErrors.rpc.invalidParams({
+          message: 'Expected a single "permittedDerivationPaths" caveat.',
+        });
+      }
+    },
+  };
+};
+
+export const getBip32EntropyBuilder = Object.freeze({
+  targetKey,
+  specificationBuilder,
+  methodHooks: {
+    getMnemonic: true,
+    getUnlockPromise: true,
+  },
+} as const);
+
+export const getBip32EntropyCaveatSpecificationBuilder: CaveatSpecificationConstraint =
+  Object.freeze({
+    type: SnapCaveatType.PermittedDerivationPaths,
+    decorator: (
+      method,
+      caveat: Caveat<
+        SnapCaveatType.PermittedDerivationPaths,
+        GetBip32EntropyParameters[]
+      >,
+    ) => {
+      return async (args) => {
+        const { params } = args;
+        validatePath(params);
+
+        const path = caveat.value.find(
+          (caveatPath) =>
+            isEqual(params.path, caveatPath.path) &&
+            caveatPath.curve === params.curve,
+        );
+
+        if (!path) {
+          throw ethErrors.rpc.invalidParams({
+            message:
+              'The requested path is not permitted. Allowed paths must be specified in the snap manifest.',
+          });
+        }
+
+        return await method(args);
+      };
+    },
+    validator: (caveat) => validateCaveatPaths(caveat),
+  });
+
+/**
+ * Builds the method implementation for `snap_getBip32Entropy`.
+ *
+ * @param hooks - The RPC method hooks.
+ * @param hooks.getMnemonic - A function to retrieve the Secret Recovery Phrase of the user.
+ * @param hooks.getUnlockPromise - A function that resolves once the MetaMask extension is unlocked
+ * and prompts the user to unlock their MetaMask if it is locked.
+ * @returns The method implementation which returns a `JsonSLIP10Node`.
+ * @throws If the params are invalid.
+ */
+function getBip32EntropyImplementation({
+  getMnemonic,
+  getUnlockPromise,
+}: GetBip32EntropyMethodHooks) {
+  return async function getBip32Entropy(
+    args: RestrictedMethodOptions<GetBip32EntropyParameters>,
+  ): Promise<JsonSLIP10Node> {
+    await getUnlockPromise(true);
+
+    // `args.params` is validated by the decorator, so it's safe to assert here.
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const derivationPath = args.params!.path!;
+
+    const node = await SLIP10Node.fromDerivationPath({
+      curve: args.params?.curve ?? 'secp256k1',
```

Comment:
Let's force people to specify this and not have any defaults. If you're using this method you need to be comfortable with specifying your curve.